connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/ConnectorUtil.java

```diff
@@ -33,11 +34,27 @@ private ConnectorUtil() {}
 
   public static Optional<OutboundConnectorConfiguration> getOutboundConnectorConfiguration(
       Class<? extends OutboundConnectorFunction> cls) {
-    return Optional.ofNullable(cls.getAnnotation(OutboundConnector.class))
-        .map(
-            annotation ->
+    Map<String, String> env = System.getenv();
+    var annotation = Optional.ofNullable(cls.getAnnotation(OutboundConnector.class));
+    if (annotation.isPresent()) {
+      final var normalizedConnectorName =
+          "CONNECTOR_" + toNormalizedConnectorName(annotation.get().name()) + "_TYPE";
+      if (env.containsKey(normalizedConnectorName)) {
+        return annotation.map(
+            ann ->
                 new OutboundConnectorConfiguration(
-                    annotation.name(), annotation.inputVariables(), annotation.type(), cls));
+                    toNormalizedConnectorName(ann.name()),
```

Comment:
Why not just remap the id instead of creation the whole new configuration like:
var type = Optional,ofNullable(env.containsKey(normalizedConnectorName)).orElse(annotation.type())

And this could be done in a separate method.

Reply:
Just trying to make it look nicer.

Reply:
> Why not just remap the id instead of creation the whole new configuration like:

I personally see this as a matter of style. If you like that one more, I can do it.

Reply:
btw, annotation here is `Optional` too, so might look a bit ugly